doc/threading_model.md

```diff
@@ -0,0 +1,31 @@
+﻿# gRPC Threading Model

```

Comment:
@LisaFC,  Please let me know what you would like to see here. The contribution note, it's location etc.